src/transformers/models/pix2struct/modeling_pix2struct.py

```diff
@@ -1546,8 +1541,12 @@ def custom_forward(*inputs):
                 present_key_value_states = present_key_value_states + (present_key_value_state,)
 
             if output_attentions:
-                all_attentions = all_attentions + (layer_outputs[2],)
-                all_cross_attentions = all_cross_attentions + (layer_outputs[3],)
+                if len(layer_outputs) >= 6:
+                    all_attentions = all_attentions + (layer_outputs[3],)
+                    all_cross_attentions = all_cross_attentions + (layer_outputs[5],)
+                else:
+                    all_attentions = all_attentions + (layer_outputs[2],)
+                    all_cross_attentions = all_cross_attentions + (layer_outputs[3],)
```

Comment:
Same here re not needing if/else logic. I don't think we need to check the length if we know `output_attentions` is passed 

```suggestion
                all_attentions = all_attentions + (layer_outputs[3],)
                all_cross_attentions = all_cross_attentions + (layer_outputs[5],)
```

Reply:
Like before, the length of layer_outputs here may not necessarily be 6, but could be 4.